src/core/public/application/ui/app_container.tsx

```diff
@@ -59,13 +64,14 @@ export const AppContainer: FunctionComponent<Props> = ({ mounter, appId }: Props
         (await mounter.mount({
           appBasePath: mounter.appBasePath,
           element: elementRef.current!,
+          onAppLeave: handler => setAppLeaveHandler(appId, handler),
         })) || null;
       setAppNotFound(false);
     };
 
     mount();
     return unmount;
-  }, [mounter]);
+  }, [appId, mounter, setAppLeaveHandler]);
```

Comment:
ESLint was showing an error about unregistered dependencies for the effect. I think that doesn't have any impact anyway as all three properties are updated at the same time